include/functions.php

```diff
@@ -0,0 +1,220 @@
+
+function echoNavAndHead($Title) {
+	echo "
+		<head>
+			<title>$Title</title>
+			<link rel='stylesheet' href='style.css'/>
+
+		</head>
+
+		<body>
+			<div class='navbarStyle'>
+				<a href='main.php'>Home |</a>
+				<a href='blogIndex.php'>Blog |</a>
+				<a href='projects.php'>Projects</a>
+			</div>
+	</body>
+";
+}
+
+
+function printNicely($interm) {
+	echo "<pre>";
+	var_dump($interm);
+	echo "<pre>";
+}
+
+
+//Blog functions!!
+
+function getAllBlogPosts() {
+	$result = dbQuery("
+		SELECT *
+		FROM blogPost
+	")->fetchAll();
+
+	return $result;
+}
+
+function getOneBlogPost($newBlogPostID) {
+	$result = dbQuery("
+		SELECT *
+		FROM blogPost
+		WHERE blogPostID = :blogPostID
+	", array(
+		'blogPostID'=>$newBlogPostID
+		))->fetch();
+
+	return $result;
+}
+
+
+//don't use this ever, probably. just use a foreach function
+function getNumPosts(){
+	$result = dbQuery("
+		SELECT COUNT(blogPostID) AS NumRows
+		FROM blogPost
+	")->fetch();
+	return $result;
+}
+
+//primary key will auto increment itself, don't need to add
+function insertBlogPost($title, $body, $dateCreated, $authorOfPost) {
+	$result = dbQuery("
+		INSERT INTO blogPost(title, body, dateCreated, authorOfPost)
+		VALUES('$title', '$body', '$dateCreated', '$authorOfPost')
+	")->fetch();
+}
+
+//deletes blog post by ID number, not any other factor
+function deleteBlogPost($newBlogPostID) {
+	$result = dbQuery("
+		DELETE FROM blogPost
+		WHERE blogPostID = :blogPostID
+		", array(
+			'blogPostID'=>$newBlogPostID
+		))->fetch();
+}
+
+
+
+//comment functions
+function getPostComments($thisBlogPostID){
+	$result = dbQuery("
+		SELECT *
+		FROM comments
+		WHERE blogPostID = :blogPostID
+		ORDER BY datePosted DESC
+		", array(
+			'blogPostID'=>$thisBlogPostID
+	))->fetchAll();
+	return $result;
+}
+
+
+function getAllTags(){
+	$result = dbQuery("
+		SELECT *
+		FROM tags
+		"
+	)->fetchAll();
+	return $result;
+}
+
+function getAllTagNames(){
+	$result = dbQuery("
+		SELECT DISTINCT tag
+		FROM tags
+		"
+	)->fetchAll();
+	return $result;
+}
+
+function getTagsWithThisName($oneTagName){
+	$result = dbQuery("
+		SELECT *
+		FROM tags
+		WHERE tag = :tag
+	", array(
+		'tag'=>$oneTagName
+	))->fetchAll();
+
+	return $result;
+}
+
+function getTagNamesWithThisNumber($oneTagNumber){
+	$result = dbQuery("
+		SELECT tag
+		FROM tags
+		WHERE blogPostID = :blogPostID
+	", array(
+		'blogPostID'=>$oneTagNumber
+	))->fetchAll();
+	return $result;
+}
+
+
+function getPostsWithThisAuthor($Author){
+	$result = dbQuery("
+		SELECT blogPostID
+		FROM blogPost
+		WHERE authorOfPost = :authorOfPost
+	", array(
+		'authorOfPost'=>$Author
+	))->fetchAll();
+	return $result;
+}
+
+function getAllAuthorNames(){
+	$result = dbQuery("
+		SELECT DISTINCT authorOfPost
+		FROM blogPost
+		"
+	)->fetchAll();
+	return $result;
+
+}
+
+
+
+function printBlogPost($var){
+
+echo "<body class='bg'>";
+	$post=getOneBlogPost($var);
+
+	echo "<div class='textStyle'>";
+		echo "<strong>Author: </strong>";
```

Comment:
This is fine, but a shorter way to write it would be:
```
echo "
   <div class='textStyle'>
      <strong>Author: </strong>
      $post[authorOfPost]
...and so on
```
I personally find all the echos to be a bit distracting and unnecessary.

Reply:
(P.S. I'm just putting a thumbs up now when I complete these changes, for my own reference. Let me know if it's annoyingly pinging you and I can stop!)

Reply:
I don't get notifications when you leave the thumbs up, so it doesn't bother me at all.